src/components/hmi_message_handler/src/mqueue_adapter.cc

```diff
@@ -48,14 +49,22 @@ class ReceiverThreadDelegate : public threads::ThreadDelegate {
   ReceiverThreadDelegate(mqd_t mqueue_descriptor,
                          HMIMessageHandler* hmi_message_handler)
       : mqueue_descriptor_(mqueue_descriptor)
-      , hmi_message_handler_(hmi_message_handler) {}
+      , hmi_message_handler_(hmi_message_handler)
+      , shutdown_requested_(false) {}
 
  private:
-  virtual void threadMain() {
-    while (true) {
+  void threadMain() OVERRIDE {
+    while (!shutdown_requested_) {
       static char buffer[kMqueueMessageSize];
-      const ssize_t size =
-          mq_receive(mqueue_descriptor_, buffer, kMqueueMessageSize, NULL);
+
+      struct timespec current_time;
+      clock_gettime(CLOCK_REALTIME, &current_time);
+
+      const time_t wait_time_sec = 1;
+      current_time.tv_sec += wait_time_sec;
+
+      const ssize_t size = mq_timedreceive(
+          mqueue_descriptor_, buffer, kMqueueMessageSize, NULL, &current_time);
```

Comment:
@anosach-luxoft When I applied similar approach in the `UsbHandlerDelegate` I was warned by @pestOO that this consumes too much CPU time as it checks the queue without pauses. So I preferred timed wait.